editor/plugins/path_3d_editor_plugin.cpp

```diff
@@ -40,17 +40,28 @@
 #include "scene/gui/menu_button.h"
 #include "scene/resources/curve.h"
 
-static bool _is_in_handle(int p_id, int p_num_points) {
-	int t = (p_id + 1) % 2;
-	int idx = (p_id + 1) / 2;
-	// order of points is [out_0, out_1, in_1, out_2, in_2, ... out_n-1, in_n-1, in_n]
-	if (idx == 0) {
-		return false;
-	} else if (idx == (p_num_points - 1)) {
-		return true;
-	} else {
-		return (t == 1);
-	}
+// Map handle id to control point id and handle type.
+enum HandleType {
+	IN = 0,
+	OUT,
+	TILT,
+};
```

Comment:
```suggestion
// Map handle id to control point id and handle type.
enum HandleType {
	IN,
	OUT,
	TILT,
};
```

This explicit index is no longer necessary if you use struct correctly.

Reply:
the `=0` is not strictly necessary per se, as that is the default behavior anyway. I add it to make it more explicit, because that makes the intention of the converting code such `info.type = HandleType(p_id % 3);` more clear.

Reply:
So you need to change to a architecture design that does not require `id % 3`.

Reply:
see the other thread. the `id % 3` part is probably unavoidable with current `EditorNode3DGizmo::add_handles()`.